compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1691,6 +1691,34 @@ private BType checkArrayType(BLangListConstructorExpr listConstructor, BArrayTyp
         return errored ? symTable.semanticError : arrayType;
     }
 
+    public boolean getLocationOfInferredArray(BType type) {
+        switch (type.tag) {
+            case TypeTags.UNION:
+                for (BType memberType : ((BUnionType) type).getMemberTypes()) {
+                    boolean pos = getLocationOfInferredArray(memberType);
+                    if (pos) {
+                        return true;
+                    }
+                }
+                return false;
+            case TypeTags.INTERSECTION:
+                for (BType memberType : ((BIntersectionType) type).getConstituentTypes()) {
+                    boolean pos = getLocationOfInferredArray(memberType);
+                    if (pos) {
+                        return true;
+                    }
+                }
+                return false;
+            case TypeTags.ARRAY:
+                if (isArrayOpenSealedType((BArrayType) type)) {
+                    return true;
+                }
+                return false;
```

Comment:
`return isArrayOpenSealedType((BArrayType) type)`

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34612/commits/f257af2b34e646f65084b82af8f62d19c94e513e.